### ISSUE TITLE
[Question] Understanding How to Use Animation

### ISSUE THREAD

**Issue by anaszakariyah** [davideas/FlexibleAdapter#298]
hello @davideas. Thank you for this wonderful library. I am using this for my project. I am still newbie on Android Development. I have a question, hopefully you don't mind me asking too much.

How to set up animation on scrolling entry? Like how Google+ doing? They use slide item from bottom.

I am tracing the code app but I have difficulty to understand it.
The [Wiki](https://github.com/davideas/FlexibleAdapter/wiki/5.x-%7C-Adapter-Animations) doesn't seem clearly documenting how to set up animation step by step.

Honestly with typical `RecyclerView.Adapter`, I usually animate the view when `onBindViewHolder(ViewHolder, int)` firstExecuting and track it by incrementing integer variable `lastAnimatedPosition` from -1 so that the animation will execute once. I wonder its possible to do this with this library?

That said, after looking through the example app, I now know that you can use ItemAnimator like how your library implement it. So, I want to learn this approach, given that looks like once it is set up, it can be easily configurable for many cases, not just on scrolling entry. Flexible, all right :)

Could you please make a step by step guide to add animation? Which one to override and implement like how you explain on [Simple Initialization Wiki](https://github.com/davideas/FlexibleAdapter/wiki/5.x-%7C-Setting-Up)?

**Comment by davideas** [comment]
@anaszakariyah, I thought that this page [Adapter-Animations](https://github.com/davideas/FlexibleAdapter/wiki/5.x-%7C-Adapter-Animations) and [AnimatorHelper](https://github.com/davideas/FlexibleAdapter/wiki/5.x-%7C-AnimatorHelper) are well done. Have you checked them?

Your mechanism is already handled by the adapter, just configure the Adapter to enable the animation as described in the Wiki and implement `scrollAnimators()` in the ViewHolder.

Basically you want the animation as G+, you can see it in action in the Overview page in my demoApp, when you display the list as LinearLayout. In the code you can check where `AnimatorHelper.slideInFromBottomAnimator(...)` is used.

**Comment by anaszakariyah** [comment]
So all you got to do is to implement AnimatedViewHolder on the ViewHolder item and set the rules using methods in [Adapter-Animations](https://github.com/davideas/FlexibleAdapter/wiki/5.x-%7C-Adapter-Animations)?

Now it does make sense.

How about when you set recyclerView.setItemAnimator() using custom ItemAnimator? Does it handled the same?

**Comment by davideas** [comment]
Almost: AnimatedViewHolder is to give unique animation to that view type only instead of using custom ItemAnimator for all view types when this is assigned to RV.
The scroll animation is achieved without AnimatedViewHolder.

**Comment by anaszakariyah** [comment]
I saw OverviewPage code. The demo is giving example by overriding` FlexibleAdapter`. I was under impression that I only have to declare and implement flexible item model and leave the adapter alone. That's why I thought that I must implement `AnimatedViewHolder` to achieve scroll animation.

**Comment by davideas** [comment]
Well, overriding the Adapter remains the alternative old option, and the call you see (`animateView()`) is manual, while with item delegation is automatic.
In both, you still need to activate animations and implement which animation you want in the VH.
